### PR TITLE
Get the MyAccount access url specific to the user tenant.

### PR DESCRIPTION
**Pull request by senthalan**

### CODE REVIEW

File: apps/authentication-portal/src/main/webapp/basicauth.jsp [hunk 1]
```diff
@@ -516,7 +516,7 @@
             }
 
             return accessURL.replaceAll(Pattern.quote("${UserTenantHint}"), userTenantHint)
-                            .replaceAll(Pattern.quote("/t/carbon.super/"), "/");
+                            .replaceAll(Pattern.quote("/t/carbon.super"), "");
```
Comment (mefarazath) [inline apps/authentication-portal/src/main/webapp/basicauth.jsp:519]: Shouldn't we use the same utility class here as well?

Comment (senthalan) [reply apps/authentication-portal/src/main/webapp/basicauth.jsp:519]: Will update it after merging this https://github.com/wso2/carbon-identity-framework/pull/3388

**Review by IsuraD — APPROVED** [review]

File: apps/recovery-portal/src/main/webapp/lite-user-confirm.jsp [hunk 2]
```diff
@@ -43,8 +43,8 @@
     String callback = request.getParameter("callback");
 
     if (StringUtils.isBlank(callback)) {
-        callback = IdentityManagementEndpointUtil.getUserPortalUrl(
-                application.getInitParameter(IdentityManagementEndpointConstants.ConfigConstants.USER_PORTAL_URL));
+        IdentityManagementEndpointUtil.getUserPortalUrl(
```
Comment (IsuraD) [inline apps/recovery-portal/src/main/webapp/lite-user-confirm.jsp:46]: Don't we need to assign to callback?

File: apps/recovery-portal/src/main/webapp/password-recovery-confirm.jsp [hunk 3]
```diff
@@ -70,8 +70,8 @@
         tenantDomain = IdentityManagementEndpointConstants.SUPER_TENANT;
     }
     if (StringUtils.isBlank(callback)) {
-        callback = IdentityManagementEndpointUtil.getUserPortalUrl(
-                application.getInitParameter(IdentityManagementEndpointConstants.ConfigConstants.USER_PORTAL_URL));
+        IdentityManagementEndpointUtil.getUserPortalUrl(
```
Comment (IsuraD) [inline apps/recovery-portal/src/main/webapp/password-recovery-confirm.jsp:73]: callback = ?

File: apps/recovery-portal/src/main/webapp/password-recovery-notify.jsp [hunk 4]
```diff
@@ -51,8 +51,8 @@
     String callback = (String) request.getAttribute("callback");
     String sessionDataKey = (String) request.getAttribute("sessionDataKey");
     if (StringUtils.isBlank(callback)) {
-        callback = IdentityManagementEndpointUtil.getUserPortalUrl(
-                application.getInitParameter(IdentityManagementEndpointConstants.ConfigConstants.USER_PORTAL_URL));
+        IdentityManagementEndpointUtil.getUserPortalUrl(
```
Comment (IsuraD) [inline apps/recovery-portal/src/main/webapp/password-recovery-notify.jsp:54]: callback = ?

File: apps/recovery-portal/src/main/webapp/password-recovery-with-claims-notify.jsp [hunk 5]
```diff
@@ -29,8 +29,8 @@
 <%
     String callback = IdentityManagementEndpointUtil.getStringValue(request.getParameter("callback"));
     if (StringUtils.isBlank(callback)) {
-        callback = IdentityManagementEndpointUtil.getUserPortalUrl(
-                application.getInitParameter(IdentityManagementEndpointConstants.ConfigConstants.USER_PORTAL_URL));
+        IdentityManagementEndpointUtil.getUserPortalUrl(
```
Comment (IsuraD) [inline apps/recovery-portal/src/main/webapp/password-recovery-with-claims-notify.jsp:32]: callback = ?

File: apps/recovery-portal/src/main/webapp/password-recovery-with-claims-options.jsp [hunk 6]
```diff
@@ -45,8 +45,8 @@
 
     String recaptchaResponse = request.getParameter("g-recaptcha-response");
     if (StringUtils.isBlank(callback)) {
-        callback = IdentityManagementEndpointUtil.getUserPortalUrl(
-                application.getInitParameter(IdentityManagementEndpointConstants.ConfigConstants.USER_PORTAL_URL));
+        IdentityManagementEndpointUtil.getUserPortalUrl(
```
Comment (IsuraD) [inline apps/recovery-portal/src/main/webapp/password-recovery-with-claims-options.jsp:48]: callback = ?

File: apps/recovery-portal/src/main/webapp/password-recovery-with-claims.jsp [hunk 7]
```diff
@@ -200,8 +200,8 @@
                             }
                             String callback = request.getParameter("callback");
                             if (StringUtils.isBlank(callback)) {
-                                callback = IdentityManagementEndpointUtil.getUserPortalUrl(
-                                        application.getInitParameter(IdentityManagementEndpointConstants.ConfigConstants.USER_PORTAL_URL));
+                                IdentityManagementEndpointUtil.getUserPortalUrl(
```
Comment (IsuraD) [inline apps/recovery-portal/src/main/webapp/password-recovery-with-claims.jsp:203]: callback = ?

File: apps/recovery-portal/src/main/webapp/recovery.jsp [hunk 8]
```diff
@@ -53,8 +53,8 @@
     String userTenant = request.getParameter("t");
 
     if (StringUtils.isBlank(callback)) {
-        callback = IdentityManagementEndpointUtil.getUserPortalUrl(
-                application.getInitParameter(IdentityManagementEndpointConstants.ConfigConstants.USER_PORTAL_URL));
+        IdentityManagementEndpointUtil.getUserPortalUrl(
```
Comment (IsuraD) [inline apps/recovery-portal/src/main/webapp/recovery.jsp:56]: callback = ?

File: apps/recovery-portal/src/main/webapp/self-registration-complete.jsp [hunk 9]
```diff
@@ -24,13 +24,14 @@
...
 <jsp:directive.include file="includes/localize.jsp"/>
+<jsp:directive.include file="tenant-resolve.jsp"/>
 <%
     boolean isEmailNotificationEnabled = false;
 
     String callback = (String) request.getAttribute("callback");
     if (StringUtils.isBlank(callback)) {
-        callback = IdentityManagementEndpointUtil.getUserPortalUrl(
-                application.getInitParameter(IdentityManagementEndpointConstants.ConfigConstants.USER_PORTAL_URL));
+        IdentityManagementEndpointUtil.getUserPortalUrl(
```
Comment (IsuraD) [inline apps/recovery-portal/src/main/webapp/self-registration-complete.jsp:33]: callback = ?

File: apps/recovery-portal/src/main/webapp/self-registration-process.jsp [hunk 10]
```diff
@@ -100,8 +100,8 @@
                                 true, false, IdentityManagementEndpointConstants.Consent.INFINITE_TERMINATION);
             }
             if (StringUtils.isBlank(callback)) {
-                callback = IdentityManagementEndpointUtil.getUserPortalUrl(
-                        application.getInitParameter(IdentityManagementEndpointConstants.ConfigConstants.USER_PORTAL_URL));
+                IdentityManagementEndpointUtil.getUserPortalUrl(
```
Comment (IsuraD) [inline apps/recovery-portal/src/main/webapp/self-registration-process.jsp:103]: callback = ?

File: apps/recovery-portal/src/main/webapp/self-registration-with-verification-confirm.jsp [hunk 11]
```diff
@@ -50,8 +50,8 @@
     }
 
     if (StringUtils.isBlank(callback)) {
-        callback = IdentityManagementEndpointUtil.getUserPortalUrl(
-                application.getInitParameter(IdentityManagementEndpointConstants.ConfigConstants.USER_PORTAL_URL));
+        IdentityManagementEndpointUtil.getUserPortalUrl(
```
Comment (IsuraD) [inline apps/recovery-portal/src/main/webapp/self-registration-with-verification-confirm.jsp:53]: callback = ?

File: apps/recovery-portal/src/main/webapp/self-registration-with-verification.jsp [hunk 12]
```diff
@@ -100,8 +100,8 @@
     }
 
     if (StringUtils.isBlank(callback)) {
-        callback = IdentityManagementEndpointUtil.getUserPortalUrl(
-                application.getInitParameter(IdentityManagementEndpointConstants.ConfigConstants.USER_PORTAL_URL));
+        IdentityManagementEndpointUtil.getUserPortalUrl(
```
Comment (IsuraD) [inline apps/recovery-portal/src/main/webapp/self-registration-with-verification.jsp:103]: callback = ?

File: apps/recovery-portal/src/main/webapp/username-recovery.jsp [hunk 13]
```diff
@@ -198,8 +198,8 @@
                             String callback = request.getParameter("callback");
 
                             if (StringUtils.isBlank(callback)) {
-                                callback = IdentityManagementEndpointUtil.getUserPortalUrl(
-                                        application.getInitParameter(IdentityManagementEndpointConstants.ConfigConstants.USER_PORTAL_URL));
+                                IdentityManagementEndpointUtil.getUserPortalUrl(
```
Comment (IsuraD) [inline apps/recovery-portal/src/main/webapp/username-recovery.jsp:201]: callback = ?

**Pull request merged**